openmc/stats/univariate.py

```diff
@@ -1312,40 +1340,56 @@ def integral(self):
         ])
 
     def clip(self, tolerance: float = 1e-6, inplace: bool = False) -> Mixture:
-        r"""Remove low-importance points from contained discrete distributions.
+        r"""Remove low-importance points / distributions
 
-        Given a probability mass function :math:`p(x)` with :math:`\{x_1, x_2,
-        x_3, \dots\}` the possible values of the random variable with
-        corresponding probabilities :math:`\{p_1, p_2, p_3, \dots\}`, this
-        function will remove any low-importance points such that :math:`\sum_i
-        x_i p_i` is preserved to within some threshold.
+        Like :meth:`Discrete.clip`, this method will remove low-importance
+        points from discrete distributions contained within the mixture but it
+        will also clip any distributions that have negligible contributions to
+        the overall intensity.
 
         .. versionadded:: 0.14.0
 
         Parameters
         ----------
         tolerance : float
-            Maximum fraction of :math:`\sum_i x_i p_i` that will be discarded
-            for any discrete distributions within the mixture distribution.
+            Maximum fraction of intensities that will be discarded.
         inplace : bool
             Whether to modify the current object in-place or return a new one.
 
         Returns
         -------
-        Discrete distribution with low-importance points removed
+        Distribution with low-importance points / distributions removed
 
         """
+        # Determine indices for any distributions that contribute non-negligibly
+        # to overall intensity
+        intensities = [prob*dist.integral() for prob, dist in
+                       zip(self.probability, self.distribution)]
+        indices = _intensity_clip(intensities, tolerance=tolerance)
+
         if inplace:
+            # Clip mixture of distributions
+            self.probability = self.probability[indices]
+            self.distribution = [self.distribution[i] for i in indices]
+
+            # Clip points from Discrete distributions
```

Comment:
If we clip lines from a bunch of `Discrete` distributions with the same tolerance as the `Mixture` we may not be respecting the `tolerance` on the total intensity of the `Mixture`. I'm assuming the main offenders here are the `Discrete` objects, but  perhaps we could merge the Discrete distributions into a single one and apply then clip to minimize this error. Or alternatively, spit out a warning if the overall intensity reduction tolerance does get exceeded. There are a few other options here as well, but I just want to raise the point.

Reply:
In my intended use case (photon source from an irradiated material), we already combine discrete distributions by default, so this really only affects things if you have low probability discrete points and low probability non-discrete distributions (usually tabular). I don't know of a good way to get around this other than to have two separate arguments as suggested by @shimwell. Personally, I don't believe it will be that helpful for a user and will just end up confusing them more. The main idea is to have a toggle to get rid of things that don't matter, and the default tolerance of 1e-6 is a good way of doing that (I would think that most users will use the default as is).

Reply:
I agree that I don't think it's particularly helpful to add additional arguments to this and that the default will be plenty good for almost all use-cases. I think I would suggest a warning in this case since it is possible to violate the tolerance.